docs/NODE_FORMAT.md

```diff
@@ -0,0 +1,36 @@
+## Corfu Node IDs
+
+This document describes the format of Corfu node identifiers (node IDs), which are used to refer to unique Corfu instances.
+Node Ids provide an extra layer of reliability in the case of a server or network reconfiguration. For example, a DNS error
+could point a Corfu client at an incorrect node. This could lead a client to read an update from the wrong system, or believe
+that it has committed an update when it is not committed. Node IDs prevent this issue by assigning each node a unique 128-bit
+identifier at startup. When clients connect to a Corfu node, they verify that the correct Node ID is presented by the node
+before interacting with it.
+
+Node IDs consist of the following components:
```

Comment:
I think it is a bit confusing in the way its written. If the node ID provides a way of uniqueness and it is "composed" of: the protocol + host name (being this an IP address) + ... and so on... With every change of IP, the node ID changes (though being the same instance)... Isn't the Node ID only the "id" component of this list you are providing? Maybe it is just a naming thing, we are using 'node Id' for both the unique part: "id" and also for the complete 'identifier': protocol://host:port/id

Reply:
Hm, yeah let me fix the docs. I renamed things to - NodeLocator when I wrote the actual code.